// Copyright Â© 2017-2023 Trust Wallet.
//
// This file is part of Trust. The full Trust copyright notice, including
// terms governing use, modification, and redistribution, is contained in the
// file LICENSE at the root of the source code distribution tree.

syntax = "proto3";

package TW.Barz.Proto;
option java_package = "wallet.core.jni.proto";

// Input parameters for calculating a counterfactual address for ERC-4337 based smart contract wallet
message ContractAddressInput {
  // Address of the contract factory
  string factory = 1;

  // Diamond proxy facets required for the contract setup
  string diamond_cut_facet = 2;
  string account_facet = 3;
  string verification_facet = 4;
  string token_receiver_facet= 5;
  string diamond_loupe_facet = 6;
  string diamond_init = 7;
  string facet_registry = 8;

  // ERC-4337 entry point
  string entry_point = 9;

  // Bytecode of the smart contract to deploy
  string bytecode = 10;

  // PublicKey of the wallet
  string public_key = 11;
}
